packages/ubuntu_desktop_installer/lib/pages/allocate_disk_space/allocate_disk_space_dialogs.dart

```diff
@@ -0,0 +1,326 @@
+double _defaultTileHeight(BuildContext context) {
+  final baseDensity = Theme.of(context).visualDensity.baseSizeAdjustment;
+  final isDense = ListTileTheme.of(context).dense;
+  return (isDense ? 48.0 : 56.0) + baseDensity.dy;
+}
+
+class _PartitionDialogLabel extends StatelessWidget {
+  const _PartitionDialogLabel(this.text, {Key? key}) : super(key: key);
+
+  final String text;
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      contentPadding: kContentPadding,
+      title: Align(
+        alignment: Alignment.topRight,
+        child: Text(text),
+      ),
+    );
+  }
+}
+
+class _RadioValueTile<T> extends StatelessWidget {
+  const _RadioValueTile({
+    Key? key,
+    required this.value,
+    required this.groupValue,
+    required this.title,
+  }) : super(key: key);
+
+  final T value;
+  final ValueNotifier<T> groupValue;
+  final Widget title;
+
+  void _setValue(T? value) => groupValue.value = value!;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<T>(
+      valueListenable: groupValue,
+      builder: (context, groupValue, child) {
+        return RadioListTile<T>(
+          title: title,
+          value: value,
+          groupValue: groupValue,
+          onChanged: _setValue,
+        );
+      },
+    );
+  }
+}
+
+class _PartitionSizeBox extends StatelessWidget {
+  const _PartitionSizeBox({
+    Key? key,
+    required this.bytes,
+    required this.unit,
+    required this.freeSpace,
+  }) : super(key: key);
+
+  final ValueNotifier<int> bytes;
+  final ValueNotifier<DataUnit> unit;
+  final int freeSpace;
+
+  void _setBytes(double value) => bytes.value = toBytes(value, unit.value);
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: Listenable.merge([bytes, unit]),
+      builder: (context, snapshot) {
+        return Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Expanded(
+              child: SpinBox(
+                value: fromBytes(bytes.value, unit.value),
+                max: freeSpace.toDouble(),
+                onChanged: _setBytes,
+              ),
+            ),
+            const SizedBox(width: kButtonBarSpacing),
+            IntrinsicWidth(
+              child: LocalizedView(
+                builder: (context, lang) {
+                  return DropdownBuilder<DataUnit>(
+                    values: DataUnit.values,
+                    selected: unit.value,
+                    onSelected: (value) => unit.value = value,
+                    itemBuilder: (context, unit, _) {
+                      return Text(unit.localize(context));
+                    },
+                  );
+                },
+              ),
+            ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+extension _PartitionUnitLang on DataUnit {
+  String localize(BuildContext context) {
+    final lang = AppLocalizations.of(context)!;
```

Comment:
No need for null checking here, as the AppLocalizations getter is non nullable since https://github.com/canonical/ubuntu-desktop-installer/commit/658a7aeca1693170baf42cc8d5b4dc03286c578f.